src/VisualStudio/Core/Def/Implementation/LanguageClient/LanguageServerClient.cs

```diff
@@ -2,81 +2,83 @@
+    [ExportMetadata("Capabilities", "WorkspaceStreamingSymbolProvider")]
+    internal sealed class LanguageServerClient : ILanguageClient
     {
+        private const string ServiceHubClientName = "ManagedLanguage.IDE.LanguageServer";
+
         private readonly IThreadingContext _threadingContext;
         private readonly Workspace _workspace;
         private readonly IEnumerable<Lazy<IOptionPersister>> _lazyOptions;
         private readonly LanguageServerClientEventListener _eventListener;
         private readonly IAsynchronousOperationListener _asyncListener;
 
-        private readonly string _serviceHubClientName;
-        private readonly string _languageServerName;
-
         /// <summary>
         /// Gets the name of the language client (displayed to the user).
         /// </summary>
-        public abstract string Name { get; }
+        public string Name => ServicesVSResources.CSharp_VB_Language_Server_Client;
 
         /// <summary>
         /// Gets the configuration section names for the language client. This may be null if the language client
         /// does not provide settings.
         /// </summary>
-        public virtual IEnumerable<string> ConfigurationSections { get; } = null;
+        public IEnumerable<string> ConfigurationSections { get; } = null;
 
         /// <summary>
         /// Gets the initialization options object the client wants to send when 'initialize' message is sent.
         /// This may be null if the client does not need custom initialization options.
         /// </summary>
-        public virtual object InitializationOptions { get; } = null;
+        public object InitializationOptions { get; } = null;
 
         /// <summary>
         /// Gets the list of file names to watch for changes.  Changes will be sent to the server via 'workspace/didChangeWatchedFiles'
         /// message.  The files to watch must be under the current active workspace.  The file names can be specified as a relative
         /// paths to the exact file, or as glob patterns following the standard in .gitignore see https://www.kernel.org/pub/software/scm/git/docs/gitignore.html files.
         /// </summary>
-        public virtual IEnumerable<string> FilesToWatch { get; } = null;
+        public IEnumerable<string> FilesToWatch { get; } = null;
 
 #pragma warning disable CS0067 // event never used - implementing interface ILanguageClient
         public event AsyncEventHandler<EventArgs> StartAsync;
         public event AsyncEventHandler<EventArgs> StopAsync;
 #pragma warning restore CS0067 // event never used
 
-        public AbstractLanguageServerClient(
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public LanguageServerClient(
             IThreadingContext threadingContext,
-            Workspace workspace,
-            IEnumerable<Lazy<IOptionPersister>> lazyOptions,
+            VisualStudioWorkspace workspace,
```

Comment:
Is the change to `VisualStudioWorkspace` to ensure the right workspace is MEF imported or because it strictly requires a `VisualStudioWorkspace`?

Reply:
Basically the former - 
1.  The language specific impl originally retrieved the VS workspace via mef - https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/CSharp/Impl/LanguageClient/CSharpLanguageServerClient.cs#L33 , so just keeping that the same.
2.  The VS workspace is exported as `VisualStudioWorkspace` , so MEF importing `Workspace` won't find anything and fail mef composition